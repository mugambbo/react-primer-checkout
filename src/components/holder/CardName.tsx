import React, { useState } from 'react'
import I18nHelper from '../../helpers/I18nHelper';
import { useI18nContext } from '../../i18n/I18nContext'
import { UserIcon } from '../svg';

interface CardNameProps {
    cardName: string;
    inputStyles?: object;
    loading: boolean;
    onCardNameChange(event: React.ChangeEvent): void
}

function CardName({loading, cardName, inputStyles, onCardNameChange}: CardNameProps) {
    const contentString = useI18nContext();
    const [focus, setFocus] = useState(false);

    const handleFocusIn = (_: React.FocusEvent) => {
        setFocus(true);
    }

    const handleFocusOut = (_: React.FocusEvent) => {
        setFocus(false);
    }

    return (
        <div id="card-name" className={`input-container my-8${focus? " input-border": ""}`} >
            <div className="input-icon">
                <UserIcon />
            </div>
            <input name="card-name" className="card-input full-width" required disabled={loading} placeholder={I18nHelper.formatString(contentString.placeholder_cardholder_name)} value={cardName} onFocus={handleFocusIn} onBlur={handleFocusOut} style={{...inputStyles}} onChange={onCardNameChange} />
        </div>
    )
}

export default CardName
