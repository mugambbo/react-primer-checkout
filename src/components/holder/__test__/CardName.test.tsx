import React from 'react';
import { mount } from 'enzyme';
import CardName from '../CardName';

describe('<CardName />', () => {
    const onCardNameChange = jest.fn();

    test('renders the card name input', () => {    
        const loading = false;
        const cardName = "Abdulmajid";
        const component = mount(<CardName cardName={cardName} loading={loading} onCardNameChange={onCardNameChange} />);  
        expect(component).toMatchSnapshot();
    });

    test('renders the card name input: changed', () => {
        const loading = false;
        const cardName = "Abdulmajid";
        const wrapper = mount(<CardName cardName={cardName} loading={loading} onCardNameChange={onCardNameChange} />);  
        const cardNameInputComponent = wrapper.find("input").first();
        cardNameInputComponent.simulate('focus').simulate('change', { target: { value: cardName } });
        cardNameInputComponent.simulate('blur');
        const cardNameInputElement: React.ReactElement<HTMLInputElement>  = cardNameInputComponent.getElement();
        expect(cardNameInputElement.props.value).toEqual(cardName);
    });
});