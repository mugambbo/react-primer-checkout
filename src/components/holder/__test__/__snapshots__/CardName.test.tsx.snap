// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardName /> renders the card name input 1`] = `
<CardName
  cardName="Abdulmajid"
  loading={false}
  onCardNameChange={[MockFunction]}
>
  <div
    className="input-container my-8"
    id="card-name"
  >
    <div
      className="input-icon"
    >
      <User>
        <svg
          enableBackground="new 0 0 216 146"
          height="16px"
          id="user"
          version="1.1"
          viewBox="0 0 216 146"
          width="24px"
          x="0px"
          xmlSpace="preserve"
          xmlns="http://www.w3.org/2000/svg"
          y="4px"
        >
          <g>
            <path
              className="svg"
              d="M107.999,73c8.638,0,16.011-3.056,22.12-9.166c6.111-6.11,9.166-13.483,9.166-22.12c0-8.636-3.055-16.009-9.166-22.12c-6.11-6.11-13.484-9.165-22.12-9.165c-8.636,0-16.01,3.055-22.12,9.165c-6.111,6.111-9.166,13.484-9.166,22.12c0,8.637,3.055,16.01,9.166,22.12C91.99,69.944,99.363,73,107.999,73z"
              fill="#393939"
            />
            <path
              className="svg"
              d="M165.07,106.037c-0.191-2.743-0.571-5.703-1.141-8.881c-0.57-3.178-1.291-6.124-2.16-8.84c-0.869-2.715-2.037-5.363-3.504-7.943c-1.466-2.58-3.15-4.78-5.052-6.6s-4.223-3.272-6.965-4.358c-2.744-1.086-5.772-1.63-9.085-1.63c-0.489,0-1.63,0.584-3.422,1.752s-3.815,2.472-6.069,3.911c-2.254,1.438-5.188,2.743-8.799,3.909c-3.612,1.168-7.237,1.752-10.877,1.752c-3.639,0-7.264-0.584-10.876-1.752c-3.611-1.166-6.545-2.471-8.799-3.909c-2.254-1.439-4.277-2.743-6.069-3.911c-1.793-1.168-2.933-1.752-3.422-1.752c-3.313,0-6.341,0.544-9.084,1.63s-5.065,2.539-6.966,4.358c-1.901,1.82-3.585,4.02-5.051,6.6s-2.634,5.229-3.503,7.943c-0.869,2.716-1.589,5.662-2.159,8.84c-0.571,3.178-0.951,6.137-1.141,8.881c-0.19,2.744-0.285,5.554-0.285,8.433c0,6.517,1.983,11.664,5.948,15.439c3.965,3.774,9.234,5.661,15.806,5.661h71.208c6.572,0,11.84-1.887,15.806-5.661c3.966-3.775,5.948-8.921,5.948-15.439C165.357,111.591,165.262,108.78,165.07,106.037z"
              fill="#393939"
            />
          </g>
        </svg>
      </User>
    </div>
    <input
      className="card-input full-width"
      disabled={false}
      name="card-name"
      onBlur={[Function]}
      onChange={[MockFunction]}
      onFocus={[Function]}
      placeholder="Cardholder Name"
      required={true}
      style={Object {}}
      value="Abdulmajid"
    />
  </div>
</CardName>
`;
